### PR TITLE
Migrate to GitHub actions

### PR DESCRIPTION
**Pull request by Crevil**
This change migrates the CI and CD pipeline to GitHub actions.
The build workflow does the same as Travis CI setup.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: build
+
+on:
+  push:
+    branches:
+      - "master"
+    tags:
+      - "v*"
+  pull_request:
```
Comment (mahlunar) [inline .github/workflows/build.yml:9]: Is this right? Should there not be a value?

**Review by mahlunar — APPROVED** [review]
This is awesome. Let's also add Dependabot support for go.mod and github actions

File: .github/workflows/build.yml (hunk 1 above)
Comment (Crevil) [reply .github/workflows/build.yml:9]: It is from their docs so I guess it's good: https://goreleaser.com/ci/actions/

**Comment by Crevil** [comment]
You gonna love #280 @mahlunar 🚀

**Comment by Crevil** [comment]
I realize that this change actually adds a new feature as well. Caching of Go modules 🚀

**Comment by mahlunar** [comment]
> I realize that this change actually adds a new feature as well. Caching of Go modules 🚀

Great!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Tests
+        run: |
+          make integration-test VERBOSE=yes
+      - name: Docker Login
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
+      - name: Run GoReleaser
```
Comment (mahlunar) [inline .github/workflows/build.yml:45]: We might be able to run go-release in a separate action if we want? To avoid cluttering this one

Comment (Crevil) [reply .github/workflows/build.yml:45]: Yeah it might make sense. I'm in unknown territory here so let's refactor as makes sense 👍

Comment (mahlunar) [reply .github/workflows/build.yml:45]: We learn as we go!

**Comment by Crevil** [comment]
Great I got a green build 🚀  #282 contains the above commits. Once that is merged I will rebase this branch so we can get actions in with a clean commit.

**Pull request merged**